polkadot/node/core/provisioner/src/disputes/prioritized_selection/mod.rs

```diff
@@ -143,7 +143,16 @@ where
 	metrics.on_fetched_onchain_disputes(onchain.keys().len() as u64);
 
 	gum::trace!(target: LOG_TARGET, ?leaf, "Fetching recent disputes");
-	let recent_disputes = request_disputes(sender).await;
+	// Filter out unconfirmed disputes. However if the dispute is already onchain - don't skip it.
+	// In this case we'd better push as much fresh votes as possible to bring it to conclusion
+	// faster.
+	let recent_disputes = request_disputes(sender)
+		.await
+		.into_iter()
+		.filter(|(key, dispute_status)| {
+			dispute_status.is_confirmed_concluded() || onchain.contains_key(key)
+		})
+		.collect::<BTreeMap<_, _>>();
```

Comment:
any reason to move filtering up here? the logs below "Filtering recent disputes" are no longer in the right place and got recent onchain disputes log now shows filtered number

Reply:
Because of a comment of mine: https://github.com/paritytech/polkadot-sdk/pull/9309#discussion_r2228467567

Maybe it wasn't a good idea after all.